Hit count breakpoints feature

digeff opened the pull request:
* Did some big architectural changes to the breakpoints code to make hit count breakpoints work

Now hit count breakpoints is built on top of "normal breakpoints". Hit count breakpoints has a "private breakpoint subsystem" that is only used to handle the hit count breakpoints.

BreakpointsUpdater calls SingleBreakpointSetterWithHitCountSupport to set breakpoints.
SingleBreakpointSetterWithHitCountSupport depending on whether it's a hit count breakpoint or normal one it delegates that task to: HitCountBreakpointsSetter or SingleBreakpointSetter.
SingleBreakpointSetter now is the top of the normal breakpoint system.
HitCountBreakpointsSetter will translate the hit count breakpoint to the underlying normal breakpoint that always hit, and set it using it's own copy of SingleBreakpointSetter (and all it's dependencies).

I also modified the DI logic to automatically add a: "__containerName" variable to all DI objects. The normal breakpoint system will say: "Breakpoints". The variables that are part of the private hit count breakpoint sytem will say: "Hit count breakpoints".

I also updated the source-map to the latest version while investigating an issue regarding source-maps mapping.

Also updated the DA to only send the continued event if the client was stopped. That will help us validate in the tests that we are not missing accidentally any stopped or continued events.

roblourens commented:
> I also updated the source-map to the latest version while investigating an issue regarding source-maps mapping.

This will break VS because it doesn't use a node version that supports webassembly, also I believe there were breaking api changes, so I'm surprised if this just works

digeff commented:
I did have to update the code a little to match the new APIs (I got the latest typing, and I fixed everything that made the code not compile).

We updated the version of Node that VS 2019 is using to 10.x so wasm should be supported.

digeff commented on src/chrome/internal/breakpoints/features/hitCountBreakpointsSetter.ts:
```diff
@@ -0,0 +1,160 @@
+    }
+
+    public setOnPausedForBreakpointCallback(onPausedForBreakpointCallback: OnPausedForBreakpointCallback): void {
+        this._singleBreakpointSetter.bpRecipeStatusChangedListeners.add(bpRecipeStatus => this.onUnderlyingBPRecipeStatusChange(bpRecipeStatus));
```
Moved
